Content.Shared/Whitelist/EntityWhitelistSystem.cs

```diff
@@ -23,6 +23,42 @@ public bool IsValid(EntityWhitelist list, [NotNullWhen(true)] EntityUid? uid)
         return uid != null && IsValid(list, uid.Value);
     }
 
+    /// <summary>
+    /// Checks whether a given entity satisfies a whitelist.
+    /// If the whitelist is null it returns null.
+    /// </summary>
+    public bool? Match(EntityWhitelist? list, [NotNullWhen(true)] EntityUid? uid)
+    {
+        if (list == null)
+            return null;
+
+        return IsValid(list, uid);
+    }
+
+    /// <summary>
+    /// Checks whether a given entity is allowed by a whitelist and not blocked by a blacklist.
+    /// If a blacklist is provided and it matches then this returns false.
+    /// If a whitelist is provided and it does not match then this returns false.
+    /// If either list is null it does not get checked.
+    /// </summary>
+    public bool IsAllowed([NotNullWhen(true)] EntityUid? uid, EntityWhitelist? blacklist = null, EntityWhitelist? whitelist = null)
```

Comment:
Change the name so that the difference between IsValid and this is more obvious. It should definitely specify that this contains a whitelist AND a blacklist and not just two whitelists or w/e

Reply:
CheckBoth?